server/src/main/java/org/elasticsearch/search/aggregations/AggregationReduceContext.java

```diff
@@ -210,7 +214,28 @@ public boolean isFinalReduce() {
 
         @Override
         protected void consumeBucketCountAndMaybeBreak(int size) {
-            multiBucketConsumer.accept(size);
+            final String aggregationName = builder().getName();
+            try {
+                multiBucketConsumer.accept(size);
+            } catch (MultiBucketConsumerService.TooManyBucketsException e) {
+                logger.error(
```

Comment:
I think both message should be warning? It fails a search request, but doesn't affect the node.
And we log circuit breaker failures as warning elsewhere already.

Reply:
I logged them as erros for two reasons:
1. they are related to an exception
2. ERROR and WARN are two different levels...if the level is set to ERROR we would not see WARN logs. It is not unusual to set log level ERROR in a production environment to manage the amount of logs...

Reply:
Also when filtering on log level in our logs I expect to have more WARN logs then ERROR logs...I am of the idea that important things as exceptions should be errors so that if we filter logs on log level we should see them not mixed with warnings. Anyway, I asked in Slack and it looks like `INFO` is the default log level.

Reply:
I think error logs should be reserved for errors that affect the shared state (e.g. unassigned shards. A exception caused by a search request usually doesn't harm a node or leave it in a bad state. So circuit breaker exception are logged as warning and so are suppressed  exception at [the rest layer](https://github.com/elastic/elasticsearch/blob/24581188be70d006d045b33525a1fe20658c49f9/server/src/main/java/org/elasticsearch/rest/RestResponse.java#L127-L131).

Also `CircuitBreakingException` doesn't return a 500 status code (returns a 429).